### PR TITLE
Bug/1707 analytics does not support moved properties

### PR DESCRIPTION
**Pull request by ryanwelcher**
## Summary

Addresses issue #1707 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/datastore/profiles.js [hunk 1]
```diff
@@ -99,33 +99,35 @@ const baseActions = {
...
 	 *
+	 * @param {string} accountID Google Analytics account ID.
 	 * @param {string} propertyID Google Analytics property ID.
 	 * @param {Object} args Profile arguments.
 	 * @param {string} args.profileName The name for a new profile.
 	 * @return {Object} Object with `response` and `error`.
 	 */
-	*createProfile( propertyID, { profileName } ) {
+	*createProfile( accountID, propertyID, { profileName } ) {
+		invariant( isValidAccountID( accountID ), 'a valid account ID is required to create a profiles.' );
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/datastore/profiles.js:109]:
```suggestion
		invariant( isValidAccountID( accountID ), 'a valid account ID is required to create a profile.' );
```

File: assets/js/modules/analytics/datastore/profiles.js [hunk 2]
```diff
@@ -26,63 +26,63 @@ import invariant from 'invariant';
...
 	},
-	reducerCallback: ( state, profile, { propertyID } ) => {
+	reducerCallback: ( state, profile, { accountID, propertyID } ) => {
 		return {
 			...state,
 			profiles: {
 				...state.profiles,
-				[ propertyID ]: [
+				[ `${ accountID }::${ propertyID }` ]: [
 					...( state.profiles[ propertyID ] || [] ),
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/datastore/profiles.js:75]: The key needs to be the same here as the one on the line above where it includes both IDs (i.e. `${ accountID }::${ propertyID }`).

File: assets/js/modules/analytics/datastore/accounts.test.js [hunk 3]
```diff
@@ -201,7 +202,9 @@ describe( 'modules/analytics accounts', () => {
 				// Properties and profiles should also have been received by
 				// this action.
 				const properties = registry.select( STORE_NAME ).getProperties( accountID );
-				const profiles = registry.select( STORE_NAME ).getProfiles( propertyID );
+				const profiles = registry.select( STORE_NAME ).getProfiles( accountID, propertyID );
+
+				//console.log( profiles );
```
[the remark this answers is not in the document]
Comment (ryanwelcher) [reply assets/js/modules/analytics/datastore/accounts.test.js:207]: Gah! Sorry about that, I've removed it!

File: assets/js/modules/analytics/datastore/profiles.js (hunk 2 above)
Comment (ryanwelcher) [reply assets/js/modules/analytics/datastore/profiles.js:75]: @aaemnnosttv this seems to be correct now. Can you confirm?

Comment (aaemnnosttv) [reply assets/js/modules/analytics/datastore/profiles.js:75]: The key on L74 is correct and the key on L75 should match 🙂

File: includes/Modules/Analytics.php [hunk 4]
```diff
@@ -1328,28 +1322,67 @@ private function get_provisioning_redirect_uri() {
...
+				$user_account_ids = wp_list_pluck( $user_accounts_properties_profiles['accounts'], 'id' );
+				foreach ( $user_account_ids as $user_account_id ) {
+					// Skip the inferred account id, that ship has sailed.
+					if ( $account_id === $user_account_id ) {
+						continue;
+					}
+					$properties = $this->get_data( 'properties-profiles', array( 'accountID' => $user_account_id ) );
+					if ( $has_property( $properties ) ) {
+						return array(
+							'accountID'  => $account_id,
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics.php:1375]: This should be the same account ID that is being iterated with.
```suggestion
							'accountID'  => $user_account_id,
```

**Pull request merged**